### PR TITLE
Fix mesh tester example and add tests

### PR DESCRIPTION
**Pull request by bmhan12**
This PR:
- Fixes failing `mesh_tester` executable
  - Bug was introduced by a change made in PR #1111 ([link to exact buggy commit](https://github.com/LLNL/axom/pull/1111/commits/d340e93be55a314853a6609c47b0b16cf21a20c9))
- Adds tests for `mesh_tester` for the different spatial indexes and RAJA policies.
- Re-enable `core_openmp_map` unit test.
  - `AXOM_USE_RAJA` was used of instead of `RAJA_FOUND` to guard test inclusion.

### CODE REVIEW

**Review by rhornung67 — APPROVED** [review]
Thanks @bmhan12

**Comment by agcapps** [comment]
Thanks for the fixes, @bmhan12 .  I see you test `RAJA_FOUND`.  I think we may also want to test `&& AXOM_USE_RAJA` to allow a user to manually turn off RAJA for Axom even though it's there.  Thoughts?

**Review by agcapps — APPROVED** [review]
Approved, but perhaps we also want to test against `AXOM_USE_RAJA`.

**Comment by rhornung67** [comment]
@agcapps doesn't RAJA_FOUND imply that AXOM_USE_RAJA is true/defined?

**Comment by bmhan12** [comment]
> I think we may also want to test && AXOM_USE_RAJA to allow a user to manually turn off RAJA for Axom even though it's there. Thoughts?

Hm, this might make things confusing, at least in terms of separation of defines for source code (`AXOM_USE_XXX`), and ones for CMake (`XXX_FOUND` , `ENABLE_XXX`, `AXOM_ENABLE_XXX`).

In these cases `AXOM_USE_RAJA` is not currently defined for CMake, so the checks fail.

**Comment by bmhan12** [comment]
> doesn't RAJA_FOUND imply that AXOM_USE_RAJA is true/defined?

Yes, relevant CMake configuration:
https://github.com/LLNL/axom/blob/e8092581482f648084d82b7d69f20cfd679c88eb/src/cmake/AxomConfig.cmake#L13-L21

**Comment by rhornung67** [comment]
@bmhan12 I looks like we use AXOM_USE_RAJA in the source code, but RAJA_FOUND in the CMakeLists.txt files. So I think what you have done makes sense. @agcapps do you agree? We should confirm with @white238

**Comment by agcapps** [comment]
> @agcapps doesn't RAJA_FOUND imply that AXOM_USE_RAJA is true/defined?

OK, so in https://github.com/LLNL/axom/blob/develop/src/cmake/AxomConfig.cmake#L16-L21, AXOM_USE_RAJA gets set if RAJA_FOUND is true or the user passed in ENABLE_RAJA.  Looking at source code, header and source files seem to test AXOM_USE_RAJA, but CMakeLists.txt files (which is what you're working on, Brian) seem to test RAJA_FOUND.

Looks like you are doing the right thing.  Sorry for the noise.

**Comment by rhornung67** [comment]
An occasional round of confusion is cleansing....

File: src/axom/quest/detail/Discretize_detail.hpp [hunk 1]
```diff
@@ -220,14 +220,15 @@ int discrSeg(const Point2D &a,
...
+        out_ptr[next_lvl + i * lvl_factor + 1] =
+          new_inscribed_prism(out_ptr[curr_lvl + i], U, R, Q, P, pa, pb);
         if(level == 0)
         {
-          out[next_lvl + i * lvl_factor + 2] =
-            new_inscribed_prism(out[curr_lvl + i], S, P, U, T, pa, pb);
+          out_ptr[next_lvl + i * lvl_factor + 2] =
+            new_inscribed_prism(out_ptr[curr_lvl + i], S, P, U, T, pa, pb);
         }
       });
```
Comment (kennyweiss) [inline src/axom/quest/detail/Discretize_detail.hpp:233]: I'm surprised it's necessary to grab a pointer to out, via `out.ptr()` before changing values in the array. 
That's certainly now how we want to use `ArrayView`.

Would it help if we add the `mutable` keyword? E.g.
```cpp
AXOM_LAMBDA(axom::IndexType i) mutable {
    out[ <some_index> ] = new_inscribed_prism(out[ <other_index> ], <params> );
    ...
}
```

We do that in a few places in axom, e.g.:
* https://github.com/LLNL/axom/blob/e8092581482f648084d82b7d69f20cfd679c88eb/src/axom/spin/tests/spin_bvh.cpp#L1404
* https://github.com/LLNL/axom/blob/e8092581482f648084d82b7d69f20cfd679c88eb/src/axom/quest/DistributedClosestPoint.hpp#L1262

File: src/tools/CMakeLists.txt [hunk 2]
```diff
@@ -69,6 +69,39 @@ if(AXOM_ENABLE_QUEST)
...
+                          --infile ${plane_data_dir}
+                          --method ${_method}
+                          --policy ${_policy}
+                )
+
+                set_tests_properties(${_testname} PROPERTIES
+                    PASS_REGULAR_EXPRESSION  "5 intersecting tri pairs")
+            endforeach()
+        endforeach()
+    endif()
```
Comment (kennyweiss) [inline src/tools/CMakeLists.txt:103]: 👍

**Review by kennyweiss — APPROVED** [review]
Thanks @bmhan12 !

File: src/axom/quest/detail/Discretize_detail.hpp (hunk 1 above)
Comment (bmhan12) [reply src/axom/quest/detail/Discretize_detail.hpp]: @kennyweiss Thanks! `mutable` works here!

File: src/tools/CMakeLists.txt [hunk 3]
```diff
@@ -69,6 +69,39 @@ if(AXOM_ENABLE_QUEST)
...
         blt_add_target_link_flags(TO mesh_tester FLAGS "-fgpu-rdc")
     endif()
+
+    # Add unit tests
+    if(AXOM_ENABLE_TESTS AND AXOM_DATA_DIR AND RAJA_FOUND AND UMPIRE_FOUND)
+
+        # Run the mesh_tester with the plane_simp_problems.stl example and
+        # different spatial indexes, raja policies
+
+        set(plane_data_dir ${AXOM_DATA_DIR}/quest/plane_simp_problems.stl)
```
Comment (white238) [inline src/tools/CMakeLists.txt:79]:
```suggestion
        set(plane_data_dir "${AXOM_DATA_DIR}/quest/plane_simp_problems.stl")
```
Should be quoted for windows if there is a space

**Review by white238 — APPROVED** [review]

File: src/tools/CMakeLists.txt (hunk 3 above)
Comment (agcapps) [reply src/tools/CMakeLists.txt]: Thanks for catching that

**Pull request merged**